homeassistant/components/izone/discovery.py

```diff
@@ -0,0 +1,133 @@
+"""Internal discovery service for  iZone AC."""
+
+import logging
+from asyncio import Event
+from typing import Dict
+
+import pizone
+
+from homeassistant.const import CONF_EXCLUDE, EVENT_HOMEASSISTANT_STOP
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+from .climate import ControllerDevice
+from .constants import DATA_ADD_ENTRIES, DATA_CONFIG, DATA_DISCOVERY_SERVICE
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_start_discovery_service(hass: HomeAssistantType):
+    """Set up the pizone internal discovery."""
+    class DiscoveryService(pizone.Listener):
+        """Discovery data and interfacing with pizone library."""
+
+        def __init__(self):
+            self.controllers = {}  # type: Dict[str, pizone.Controller]
```

Comment:
Does Listener have a constructor? You might want to call the super constructor here.

Reply:
It's just an abstract class, but can do for consistency.